plugin/restore/mysql/mysql.go

```diff
@@ -448,6 +572,45 @@ func (r *Restore) getLatestBinlogFileMeta(ctx context.Context) (*mysql.BinlogFil
 	return nil, fmt.Errorf("cannot find latest binlog on instance")
 }
 
+// getBinlogNamePrefix returns the prefix of binlog file name.
+func (r *Restore) getBinlogNamePrefix(ctx context.Context) (string, error) {
+	db, err := r.driver.GetDbConnection(ctx, "")
+	if err != nil {
+		return "", err
+	}
+
+	rows, err := db.QueryContext(ctx, `SHOW VARIABLES LIKE "log_bin_basename";`)
+	if err != nil {
+		return "", fmt.Errorf("cannot query log_bin_basename on mysql server, error: %w", err)
+	}
+	defer rows.Close()
+
+	var basename string
+	if rows.Next() {
+		var unused interface{}
+		if err := rows.Scan(&unused /*Variable_name*/, &basename); err != nil {
+			return "", err
+		}
+		return filepath.Base(basename) + ".", nil
+	}
+	return "", fmt.Errorf("cannot find log_bin_basename on mysqlbin server, error: %w", err)
+}
+
+// getBinlogSeqNumber returns the sequence number of binlog file.
+// For example: ("binlog.000001","binlog.") => 1, ("binlog.000001a", "binlog.") => err, ("binlog.000001", "binlog") => err
+func getBinlogFileNameSeqNumber(name, prefix string) (int64, error) {
```

Comment:
https://dev.mysql.com/doc/refman/8.0/en/binary-log.html
> [mysqld](https://dev.mysql.com/doc/refman/8.0/en/mysqld.html) appends a numeric extension to the binary log base name to generate binary log file names.

Let's call it ext or extension instead of seq or sequence.

https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/mysql/mysql-server%24+MAX_LOG_UNIQUE_FN_EXT&patternType=regexp

The max extension is 2147483647 in MySQL 8.0, so we should just use int for it.

Reply:
SG. I  previously focus on from 999999  to 1000000.